src/ctap/mod.rs

```diff
@@ -521,9 +546,11 @@ where
             None => return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_RESPONSE_CANNOT_WRITE_CBOR),
         };
         auth_data.extend(cose_key);
-        if use_hmac_extension {
-            let extensions = cbor_map! {
-                "hmac-secret" => true,
+        if has_extension_output {
+            let hmac_secret_output = if use_hmac_extension { Some(true) } else { None };
+            let extensions = cbor_map_options! {
+                "hmac-secret" => hmac_secret_output,
+                "credProtect" => cred_protect_policy.map(|policy| policy as i64),
```

Comment:
What is the implication of having credProtect on CTAP2.0?
Currently CTAP2.1 is gated by a rust feature and credProtect is part of it and wasn't there for 2.0, right?

Reply:
From what I heard, this extension only barely didn't make it into 2.0 (without being an extension). I don't know if extensions need to be officially mentioned in a specification to be valid. Or if I could just invent an extension and use it. This one is described in 2.1, so we could hide it behind the feature flag, if you want to. That being said, it is perfectly compatible with 2.0 and I didn't see why I wouldn't allow 2.0 users to use that extension. With the current default policy, it does not change any behavior.

Your call.

Reply:
I would say it depends how far we are at having all CTAP2.1 implemented (assuming that once fully implemented, we'll remove the feature and only be CTAP2.1) and whether supporting this extension would break OpenSK when compiled without CTAP2.1 in the meantime.

Reply:
2.1 is not finalized yet, and I still expect some commands to change in the next months. We are still missing implementations for bigger work items like enterprise attestation. credProtect should be stable and is independent of other new features. It only interacts with old commands (MakeCredential, GetAssertion, GetInfo).